src/modules/townhalls/townhalls.ts

```diff
@@ -66,8 +72,46 @@ export async function getBillInfo(townhallId: string) {
     const townhall = await Collections.Townhalls().findOne({
         _id: new ObjectID(townhallId),
     });
+
     if (!townhall) throw new Error('Invalid Townhall ID');
 
-    // eventually after doing some other requests
-    return {};
+    const subject = await getSubjectUrl(townhall?.form.topic);
+    const billUrls: string[] = await getBillUrls(subject);
+    const allBills: Bill[] = [];
+    for (let i = 0; i < billUrls.length; i += 1) {
+        // eslint-disable-next-line no-await-in-loop
+        allBills.push(await getBill(billUrls[i]));
+    }
+
+    for (let i = 0; i < allBills.length; i += 1) {
+        let houseVote = '';
+        let senateVote = '';
+        const houseObj = allBills[i].votes.find((o) => o.chamber === 'House');
+        const senateObj = allBills[i].votes.find((o) => o.chamber === 'Senate');
+        if (houseObj !== undefined) {
+            // eslint-disable-next-line no-await-in-loop
+            houseVote = await getVoteResult(
+                houseObj?.api_url,
+                townhall?.form.speaker
+            );
+        }
+
+        if (senateObj !== undefined) {
+            // eslint-disable-next-line no-await-in-loop
+            senateVote = await getVoteResult(
+                senateObj?.api_url,
+                townhall?.form.speaker
+            );
+        }
+
+        if (houseVote !== '') {
+            allBills[i].vote_position = houseVote;
+        } else if (senateVote !== '') {
+            allBills[i].vote_position = senateVote;
+        } else {
+            allBills[i].vote_position = 'Not Found';
+        }
+    }
```

Comment:
Same thing from my previous comment could be applied here regarding the await in for loop's.